lib/operations/collection/get.js

```diff
@@ -4,51 +4,73 @@ const PAGE_AFTER = 'pageAfter';
 const PAGE_NUM = 'pageNum';
 const PAGE_SIZE = 'pageSize';
 
+function isNumeric(val) {
+	let retval = false;
+	const valParsed = parseInt(val, 10);
+	if (valParsed || valParsed === 0) {
+		retval = true;
+	}
+	return retval;
+}
+
 function takeProperty(obj, prop) {
 	let retval;
 	if (obj && obj.hasOwnProperty(prop)) {
 		retval = obj[prop];
 		delete obj[prop];
-		// If the property is a numeber stored as a string we convert it to an actual number
-		if (!isNaN(retval)) {
+		// If the property is a number stored as a string we convert it to an actual number
+		if (isNumeric(retval)) {
 			retval = parseInt(retval, 10);
 		}
 	}
 	return retval;
 }
 
+function getPagination (context, map, keys, pageSize, pageAfter, pageNum) {
+	const pagination = {
+		startIndex: 0,
+		endIndex: keys.length
+	};
+	if (typeof (pageSize) !== 'undefined') {
+		if (pageSize <= 0 || typeof (pageSize) !== 'number') {
+			context.setStatus(400); // Not a number
+		} else if (typeof (pageAfter) !== 'undefined') {
+			if (map[pageAfter]) {
+				pagination.startIndex = keys.indexOf(pageAfter) + 1;
+			} else {
+				context.setStatus(400); // Key not found
+			}
+		} else if (typeof (pageNum) !== 'undefined') {
+			if (pageNum <= 0 || typeof (pageNum) !== 'number') {
+				context.setStatus(400); // Not a number
+			} else {
+				pagination.startIndex = (pageNum - 1) * pageSize;
+			}
+		}
+
+		// Only use the calculated page end index if it's smaller than the number of keys
+		const pageEndIndex = pagination.startIndex + pageSize;
+		if (pagination.endIndex > pageEndIndex) {
+			pagination.endIndex = pageEndIndex;
+		}
+	}
+	return pagination;
+}
+
 function createMap(map, context) {
 	const keys = Object.keys(map);
 	const query = context.query;
 	const result = {};
 
 	// Get special query string properties and remove them from the query object.
+	const pageSize = takeProperty(query, PAGE_SIZE);
 	const pageAfter = takeProperty(query, PAGE_AFTER);
 	const pageNum = takeProperty(query, PAGE_NUM);
```

Comment:
I could have calculated pageSize, pageAfter, and pageNum inside the getPagination function, but I decided to get them here. This is because I wanted to make it easier to see that these properties are being removed from the 'query' object before it is accessed in the loop.